app/src/main/java/com/example/userinfo/UserInputActivity.kt

```diff
@@ -0,0 +1,164 @@
+const val Email="email"
+const val PhoneNumber="phoneNumber"
+const val Address="address"
+const val PinCode="pinCode"
+const val Visible="visible"
+class UserInputActivity : AppCompatActivity() {
+
+    var visible:Boolean?= true
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_user_input)
+
+        val validateButton = findViewById<Button>(R.id.validateButton)
+        val userName = findViewById<EditText>(R.id.userNameET)
+         val email = findViewById<EditText>(R.id.emailET)
+         val phoneNumber = findViewById<EditText>(R.id.phoneNumberET)
+         val pinCode = findViewById<EditText>(R.id.pinCodeET)
+         val address = findViewById<EditText>(R.id.addressET)
+         val confirmButton= findViewById<Button>(R.id.ConfirmButton)
+         val cancelButton=findViewById<Button>(R.id.cancelButton)
+
+
+        validateButton?.setOnClickListener {
+            val allFields = ArrayList<EditText?>()
+            allFields.add(userName)
+            allFields.add(email)
+            allFields.add(phoneNumber)
+            allFields.add(pinCode)
+            allFields.add(address)
+            val allButtons=ArrayList<Button?>()
+            allButtons.add(confirmButton)
+            allButtons.add(validateButton)
+            allButtons.add(cancelButton)
+            val isAllFieldsChecked = checkAllFields(allFields)
+            if (isAllFieldsChecked) {
+                val isAllFieldsEnteredProperly = checkAllFieldsEnteredProperly(userName, email, phoneNumber, pinCode, address)
+                if (isAllFieldsEnteredProperly) {
+                    visible=false
+                    changeVisibilities(visible,userName,email,phoneNumber,pinCode,address,cancelButton,validateButton,confirmButton)
+                }
+            }
+        }
+        cancelButton?.setOnClickListener{
+            visible=true
+            changeVisibilities(visible,userName,email,phoneNumber,pinCode,address,cancelButton,validateButton,confirmButton)
+        }
+        confirmButton?.setOnClickListener{
+            val intent= Intent(this,DisplayUserInfo::class.java)
+            intent.putExtra(UserName,userName?.text.toString())
+            intent.putExtra(Address,address?.text.toString())
+            intent.putExtra(PinCode,pinCode?.text.toString())
+            intent.putExtra(PhoneNumber,phoneNumber?.text.toString())
+            intent.putExtra(Email,email?.text.toString())
+            startActivity(intent)
+        }
+    }
+
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        outState.putBoolean(Visible, visible!!)
+        super.onSaveInstanceState(outState)
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+        visible=savedInstanceState.getBoolean(Visible)
+        val validateButton = findViewById<Button>(R.id.validateButton)
```

Comment:
Normally we should initialise every time we need a view. because findViewById takes time